honeybee/_hbanalysissurface.py

```diff
@@ -78,16 +82,25 @@ def from_json(cls, srf_json):
         """Create a surface from json object.
 
         The minimum schema is:
-        {"name": "",
-        "vertices": [[(x, y, z), (x1, y1, z1), (x2, y2, z2)]],
-        "surface_type": null  // 0: wall, 5: window
-        }
+            {"name": "",
+            "vertices": [[(x, y, z), (x1, y1, z1), (x2, y2, z2)]],
+            "surface_material": {}  // radiance material json file
+            }
         """
         name = srf_json["name"]
         vertices = srf_json["vertices"]
-        type_id = srf_json["surface_type"]
-        srf_type = surfacetype.SurfaceTypes.get_type_by_key(type_id)
-        return cls(name, vertices, srf_type)
+        HBsrf = cls(name, vertices)
+        # Check material type and determine appropriate "from_json" classmethod
+        material_json = srf_json["surface_material"]
+        material_type = material_json["type"]
+        if material_type == "plastic":
+            radiance_material = PlasticMaterial.from_json(material_json)
+        elif material_type == "metal":
+            radiance_material = MetalMaterial.from_json(material_json)
+        elif material_type == "glass":
+            radiance_material = GlassMaterial.from_json(material_json)
```

Comment:
You need a final `else` here to either set it to some default material or raise an error. What is user makes a mistake for input `material_type`?

Reply:
Have added else: raise ValueError. I assumed that would be the most fitting error in case this breaks because it does not recognize the right material type.